src/app/application/components/route-view/RouteView.vue

```diff
@@ -140,10 +139,7 @@ const routeUpdate = (params: Record<string, string | undefined>) => {
   }
   routerPush(newParams)
 }
-const routeReplace = (...args: Parameters<typeof router['push']>) => {
-  router.push(...args)
-}
```

Comment:
I am unsure about removing this, I need to take another look at this again though.

Ideally `RouteView:route:update` should update the parameters of the route you are on as a user, and `RouteView:route:replace` should replace the route entirely with a differently named route. i.e. if as a user I'm on `service-summary-view` but as an engineer coding in the `service-list-view` file then `update(props)` should leave the user on the same URL pathname but update the query params on that URL

With that being said, below you are using `update` to change the route entirely back from the summary-view to the list view, whereas the intention is you should use `replace` to do this, i.e. replace the route with an entirely differently named one.

I think the fact that what you do below works but is a bug in the implementation that we haven't realized just yet because we haven't needed this usecase before now.

Maybe wrote this a bit early, but wanted to at least express a bit of uncertainty about whether we should remove this.

Reply:
How’s what I’m doing behaviorally different?

Reply:
So here:

```javascript
@tray-close="route.update({
  page: route.params.page,
  size: route.params.size,
  })"
```
Given `update` is meant to "update" the route the user is on with new params/query params (not change the route), the code above is using `update` to change the route from the summary route back to the list view, not solely update the params.

I think the code works now because `update` is slightly wrong (i.e. theres a bug there). I would prefer this code above to use `replace` (to "replace" the summary view with the list view), and then we would fix the bug with `update` to always use the route the user is on rather than the route where you are writing the code.

Going with the intention of how `update` should work, if it was fixed to be like that and you were to call `update` here instead of `replace`, you would stay on the summary view and only update the params, not return to the listing view.

Does that help?

Reply:
So, just so I’m understand what you’d change in RouteView. You would use `route.name` here, right?

```ts
const routerPush = beforePaint((params: Record<string, string | undefined>) => {
  router.push({
    name: route.name as string, // instead of `props.name`
    query: cleanQuery(params, route.query),
  })
  newParams = {}
})
```

Reply:
Umm possibly, if that makes it work as per what I explained before yeah, I think that would make it work as explained. Does that effect anything we've already done?

I think my main thing here is to use `replace` to replace the entire route, the functionality of `update` can be sorted later if need be. I'd be careful tweaking it as part of a feature PR if you were to tweak it here.

Reply:
I was just asking what you’d change so I know what its impact is. I don’t know if it breaks something we already built.

I’ve removed the removal of this code and switched to `route.replace`. I’m still not totally convinced this makes sense. I still don’t like the name `replace` for an action that doesn’t do a `router.replace`. It’s just too close.

Reply:
> I was just asking what you’d change so I know what its impact is. I don’t know if it breaks something we already built.

Ah gotcha, we are on the same page then. tbh I'm not super sure either I'd have to take a closer look to be sure.

> I’ve removed the removal of this code and switched to route.replace. I’m still not totally convinced this makes sense. I still don’t like the name replace for an action that doesn’t do a router.replace. It’s just too close.

If there's another word we can use that means the same thing ("replace the current route with an entirely new one") we could change that also.

We can do that separate tho, let me look at the changeset here again now you've made these changes, brb